### ISSUE TITLE
Timeout with print in progress with multiple "tools" (Prusa MMU2S)

### ISSUE THREAD

**Issue by StarWitch** [tjjfvi/OctoPrint-BetterHeaterTimeout#6]
I have a Prusa MK3S with the Multi-Material Unit (2S). The MMU2S requires that multiple extruders/"tools" be set up in the OctoPrint interface, under Printer Profiles. Only one "tool" (i.e: filament) is used at a time, and "shared nozzle" is selected.

I discovered an issue where I'd paused a print with OctoPrint, and then this script kicked in, killing the temperature after 600 seconds and saying the "tool timed out" on the display. This also interrupted the GCode and effectively trashed the print, causing the printer to home itself in the middle of the bed. (It wasn't too important, don't worry.)

I haven't investigated it yet but my educated guess is that it thinks that the other four tools aren't "in use" and kills the print based on their statuses.

In summary:
1. OctoPrint set up with multiple extruders/"tools" for multimaterial printing, handled via the Slicer's GCode.
2. Job is paused for a time over the limit of the timeout (600s by default).
3. BetterHeaterTimeout doesn't accurately detect that the print is paused, and kills the print.

Theory:
1. BHT can't handle multiple "tools" and decides that because the other "tools" aren't in use, it kills the print.

**Comment by StarWitch** [comment]
To add onto this, the error message on the print was "tool1 has timed out." The "tools"/filaments are labeled tool0 through tool4, and I was only doing a single colour print on tool0/the first filament in the MMU2S.

So the code definitely needs to be able to detect if at least one "toolX" is in use, not if all five are in use at once (which is impossible).

**Comment by tjjfvi** [comment]
@StarWitch I don’t think the problem is the MMU2(S), rather the pausing of the print. I’ll look into this tomorrow and see how it handles pauses.
